covid/transform.py

```diff
@@ -644,10 +645,25 @@ def transform_covidtracking_data(covidtracking_df):
 
         ###### Calculate criteria category 2. ######
         # For the criteria, we must add positive to negative tests to get the total (discarding inconclusive).
-        covidtracking_df.loc[(state,), NEW_TESTS_TOTAL_FIELD] = (
-            covidtracking_df.loc[(state,), NEW_CASES_POSITIVE_SOURCE_FIELD]
-            + covidtracking_df.loc[(state,), NEW_CASES_NEGATIVE_SOURCE_FIELD]
-        ).values
+        new_cases_positive_series = covidtracking_df.loc[
+            (state,), NEW_CASES_POSITIVE_SOURCE_FIELD
+        ]
+        new_cases_negative_series = covidtracking_df.loc[
+            (state,), NEW_CASES_NEGATIVE_SOURCE_FIELD
+        ]
+
+        # Note: `covidtracking.com` has returned `0` values for these signals from Hawaii in the past when they actually
+        #   just were missing data; to ensure that these don't affect downstream calculations adversely, fill those with
+        #   `nan`s.
+        if state == "Hawaii":
+            covidtracking_df.loc[(state,), NEW_TESTS_TOTAL_FIELD] = (
+                new_cases_positive_series.replace(0, np.nan)
```

Comment:
@smacpher so there are no valid days with 0 new positive/negative cases?

If so, this LGTM. Otherwise maybe there's a more explicit way we can say "data from Hawaii is unknown after X date" like masking all data as `nan` after a certain date (unless the 0's mixed in are dispersed throughout discontiguous time periods). Does that make sense?

Reply:
@patricksheehan I was thinking about doing that but wanted to have this automatically start working when they started reporting non-zero data. I'm going to update this to only include the negative series (which usually ranges in the 1000s+) to be safe though since that was the only one that has reported 0s since October 8th.

Reply:
Got it. I guess I'm still a bit confused why we wouldn't just mask based on time (ie set the negative series to `nan` after October 8th, continuing into the future) until we get word from CTP that the data issue is fixed. But I have not dug in!